### ISSUE TITLE
Remove setuptools dependency

### ISSUE THREAD

**Issue by mdickinson** [enthought/envisage#515]
If possible, we should remove the runtime dependency on `setuptools`, at least on the most recent Python versions.

**Comment by mdickinson** [comment]
This doesn't look feasible - yet. In particular, the `EggPluginManager` and `EggBasketPluginManager` depend on `setuptools`. (See also https://github.com/enthought/envisage/discussions/521.)

Removing from the milestone, but leaving this open, since it _would_ be good to eventually remove the dependency on `setuptools`.

**Comment by mdickinson** [comment]
Depends on #439

**Comment by mdickinson** [comment]
Closing. This was addressed in #599 and #600. (Note that we're still using `setuptools` as a build dependency. That could change, too, but that's a separate issue.)
